crates/wasmtime/src/store.rs

```diff
@@ -809,6 +816,41 @@ impl<T> Store<T> {
         self.inner
             .out_of_fuel_async_yield(injection_count, fuel_to_inject)
     }
+
+    /// Set the epoch-based-interruption refill delta.
+    ///
+    /// When epoch interruption (see [`Config::epoch_interruption()`])
+    /// is not enabled, this is a no-op. Also, if the store is not
+    /// async, this is a no-op.
+    ///
+    /// When epoch interruption is enabled, every time we yield
+    /// control to guest code, we set a deadline equal to the current
+    /// epoch tick (see [`Engine::increment_epoch()`]) plus this
+    /// delta. Sometime after at least this many ticks, if guest code
+    /// is still running, it will yield back to the event loop. Upon
+    /// re-entering (when the future is next polled), it will again
+    /// get at least this many ticks.
+    ///
+    /// The code instrumentation that implements the interruption is
+    /// careful to bound runtime, but the bound is still not
+    /// completely tight: it is limited by the maximum straight-line
+    /// function body size. Interruption checks (to see whether the
+    /// current epoch tick has reached the deadline) are performed at
+    /// every loop backedge and at every function entry.
+    ///
+    /// By default, when epoch-based interruption is enabled and if
+    /// this value is not otherwise set, the delta is one tick; that
+    /// is, whenever the epoch changes, yields will occur.
+    #[cfg(feature = "async")]
+    pub fn set_epoch_deadline_refill_delta(&mut self, delta: u64) {
```

Comment:
The naming here and with `refill_epoch_deadline` is a little confusing to me, perhaps something like:

* `set_epoch_yield_delta` - for this function
* `set_epoch` - the below function

and along the lines of a different comment if `set_epoch` is called (which should be available without the `async` feature I think) and `set_epoch_yield_delta` isn't called then we can trap on the first epoch change)

Reply:
I've renamed the methods to `set_epoch_deadline` for the "set deadline to N ticks in the future" action (analogous to `add_fuel`), and added `epoch_deadline_trap` and `epoch_deadline_async_yield_and_update` to configure the two different behaviors when deadline is reached.